### PR TITLE
group normalization layer test

### PR DESCRIPTION
**Pull request by autoih**
Address [#731 ](https://github.com/tensorflow/addons/issues/731)

### CODE REVIEW

**Comment by autoih** [comment]
Hi @Squadrick, can you please help review when you have time? Thank you.

File: tensorflow_addons/layers/normalizations_test.py [hunk 1]
```diff
@@ -246,6 +246,95 @@ def test_groupnorm_conv(self):
...
+        out -= tf.keras.backend.eval(norm.beta)
+        out /= tf.keras.backend.eval(norm.gamma)
+
+        np.testing.assert_allclose(out.mean(), 0.0, atol=1e-1)
+        np.testing.assert_allclose(out.std(), 1.0, atol=1e-1)
+
+    def test_groupnorm_different_groups(self):
+        norm1 = GroupNormalization(axis=1, groups=2, input_shape=(10, 6))
+        norm2 = GroupNormalization(axis=1, groups=1, input_shape=(10, 6))
+        norm3 = GroupNormalization(axis=1, groups=10, input_shape=(10, 6))
```
Comment (Squadrick) [inline tensorflow_addons/layers/normalizations_test.py]: Rather than repeating code for each value of `group`, you can use parameterized tests from absl. [Example usage](https://github.com/tensorflow/addons/blob/49637c3bc8c53b11a2a6490048ef824cf4c46b44/tensorflow_addons/layers/wrappers_test.py#L87).

File: tensorflow_addons/layers/normalizations_test.py [hunk 2]
```diff
@@ -246,6 +246,95 @@ def test_groupnorm_conv(self):
...
+        model = tf.keras.models.Sequential()
+        model.add(norm1)
+        # centered and variance are  5.0 and 10.0, respectively
```
Comment (Squadrick) [inline tensorflow_addons/layers/normalizations_test.py]: Remove double space in comment.

**Comment by gabrieldemarmiesse** [comment]
@googlebot I consent

**Comment by gabrieldemarmiesse** [comment]
@autoih

File: tensorflow_addons/layers/normalizations_test.py (hunk 1 above)
Comment (autoih) [reply tensorflow_addons/layers/normalizations_test.py]: Thanks @Squadrick.

**Comment by autoih** [comment]
Thanks @gabrieldemarmiesse for the reminder. Also, can you please guide me to fix error?

**Comment by gabrieldemarmiesse** [comment]
It seems it was just an internet connection issue. I'll restart the tests and hopefully it will work.

**Comment by autoih** [comment]
Thanks @gabrieldemarmiesse.

File: tensorflow_addons/layers/normalizations_test.py [hunk 3]
```diff
@@ -246,6 +246,81 @@ def test_groupnorm_conv(self):
...
+        model = tf.keras.models.Sequential()
+        norm = GroupNormalization(input_shape=(10,), groups=2)
+        model.add(norm)
+        model.compile(loss="mse", optimizer="rmsprop")
+
+        x = np.random.normal(loc=5.0, scale=10.0, size=(1000, 10))
+        model.fit(x, x, epochs=5, verbose=0)
+        out = model.predict(x)
+        out -= tf.keras.backend.eval(norm.beta)
+        out /= tf.keras.backend.eval(norm.gamma)
```
Comment (Squadrick) [inline tensorflow_addons/layers/normalizations_test.py]: You've used both `self.evaluate` and `tf.keras.backend.eval`. Stick to the former throughout the PR.

File: tensorflow_addons/layers/normalizations_test.py [hunk 4]
```diff
@@ -246,6 +246,81 @@ def test_groupnorm_conv(self):
...
+            # centered and variance are 5.0 and 10.0, respectively
+            model.compile(loss="mse", optimizer="rmsprop")
+            x = np.random.normal(loc=5.0, scale=10.0, size=(1000, 10, 3))
+            model.fit(x, x, epochs=5, verbose=0)
+            out = model.predict(x)
+            out -= np.reshape(self.evaluate(norm.beta), (1, 10, 1))
+            out /= np.reshape(self.evaluate(norm.gamma), (1, 10, 1))
+
+            self.assertAllClose(
+                out.mean(axis=(0, 1), dtype=np.float32), (0.0, 0.0, 0.0), atol=1.1e-1
```
Comment (Squadrick) [inline tensorflow_addons/layers/normalizations_test.py]: `1,1e-1` seems oddly specific, any reason for that? Just curious. Does it fail for `1e-1`?

Comment (autoih) [reply tensorflow_addons/layers/normalizations_test.py]: Thanks @Squadrick.

File: tensorflow_addons/layers/normalizations_test.py (hunk 3 above)
Comment (autoih) [reply tensorflow_addons/layers/normalizations_test.py]: Thanks @Squadrick. Do you also suggest using seed for tf.random, like [here](https://github.com/tensorflow/addons/blob/eb77ce281997305703183204fac266f839b7e75f/tensorflow_addons/optimizers/lookahead_test.py#L110)?

Comment (Squadrick) [reply tensorflow_addons/layers/normalizations_test.py]: Yup, that too. Forgot to mention it.
